### PR TITLE
Adds broken links checks

### LINKED ISSUE #204: Add hyperlink validation to spiffe.io
I'm wondering if there's a way to add hyperlink validation to spiffe.io so we don't check in pages with broken hyperlinks.

A person attempting to check in a file to spiffe.io could be notified if there's a bad link in the file to another web page. The error could indicate the line number of the broken link and the bad link itself. Or the validation could occur during the merge request. Maybe there's a way to implement this in the CI/CD process on GitHub.

### PR DESCRIPTION
**Pull request by mchurichi**
This PR introduces a new make target that helps contributors checking for broken links on the website.

The tool can be run with `make check-links` or `make docker-check-links` and will search on the local dev web server (the one you should have running at http://localhost:1313) for links that point to sites that no longer exists (404s) and report them as errors on the console, and sites that exist but have an invalid anchor (e.g. google.com/#invalid-anchor) and report them as warnings.

Additionally, the same tool is run on Netlify's CI/CD infrastructure whenever a new PR is submitted/merged and will report the results on Netlify's build logs.

Errors or warnings reported by this tool won't block the ability to commit changes or merge a PR, they will be informative only, since we only want to encourage people to fix these broken links, not block contributions.

Closes #204 

Signed-off-by: Maximiliano Churichi <maximiliano.churichi@hpe.com>

### CODE REVIEW

**Review by Andres-GC — APPROVED** [review]
LGTM

**Comment by sanderson042** [comment]
When running this new link checker, use the `-with-releases` form of the build script, so:

If you're running Docker:
```make docker-serve-with-releases```

If you're building locally:
```make serve-with-releases```

When run without `-with-releases`, the link checking output will include a lot of 404s for links to pages on GitHub (these include `LATEST-tag` in the URL).

Using the `-with-releases` form will be necessary until that form becomes the default in a future PR. Due to an enhancement in PR #214, `-with-releases` can become the default.

**Review by sanderson042 — APPROVED** [review]
I tested both the Docker and local versions of the new link checker tool. Both versions successfully found some bad links I inserted and also some existing bad links that we need to fix. :-) :-(

I wasn't able to test the ability of this tool to check new merges and add to the Netlify logs because this PR would need to be merged first (AFAIK). I'll try to check that once this is merged.

Although it is approved now, this PR should be merged next week and not now before the weekend.

**Comment by mchurichi** [comment]
> When running this new link checker, use the `-with-releases` form of the build script, so:
> 
> If you're running Docker:
> `make docker-serve-with-releases`
> 
> If you're building locally:
> `make serve-with-releases`
> 
> When run without `-with-releases`, the link checking output will include a lot of 404s for links to pages on GitHub (these include `LATEST-tag` in the URL).
> 
> Using the `-with-releases` form will be necessary until that form becomes the default in a future PR. Due to an enhancement in PR #214, `-with-releases` can become the default.

Thanks for the review, and good catch!. I added a few words in the README to clarify this: https://github.com/spiffe/spiffe.io/pull/220/commits/b39e42eed8fd9923462a06ef37a4dff8cdf9057f

File: README.md [hunk 1]
```diff
@@ -66,6 +66,28 @@ Web Server is available at //localhost:1313/ (bind address 0.0.0.0)
 
 The website is now available at [`http://localhost:1313`](http://localhost:1313). Changes in the Markdown files or the [external content descriptor file](./external.yaml) trigger a rebuild of the site. After the rebuild, the site is reloaded in your browser.
 
+### Avoiding broken links
+
+It is common that URLs you are pointing to get deprecated or moved somewhere else over time, leading to broken links on our website.
+
+In order to avoid this, there is a tool that lets you check whether there are broken links in the whole website or not.
+
+First, make sure you are serving the website locally using the `-with-releases` form of the script (`make docker-serve-with-releases` or `make serve-with-releases`), and that it is accesible at `http://localhost:1313`, then run the following command:
```
Comment (sanderson042) [inline README.md:75]: Typo: "accessible" (but correct in Spanish ;-) )

```suggestion
First, make sure you are serving the website locally using the `-with-releases` form of the script (`make docker-serve-with-releases` or `make serve-with-releases`), and that it is accessible at `http://localhost:1313`, then run the following command:
```

File: README.md [hunk 2]
```diff
@@ -66,6 +66,28 @@ Web Server is available at //localhost:1313/ (bind address 0.0.0.0)
...
+
+```shell
+make docker-check-links # if you prefer to use Docker
```
Comment (sanderson042) [inline README.md:78]: Maybe this change will make the steps slightly more clear:

```suggestion
make docker-check-links # if you are using Docker to serve the website
```

File: README.md [hunk 3]
```diff
@@ -66,6 +66,28 @@ Web Server is available at //localhost:1313/ (bind address 0.0.0.0)
...
+```
+
+or
+
+```shell
+make check-links # if you prefer to run the toolchain in your machine
```
Comment (sanderson042) [inline README.md:84]:
```suggestion
make check-links # if you are using a local toolchain to serve the website
```

File: README.md [hunk 4]
```diff
@@ -66,6 +66,28 @@ Web Server is available at //localhost:1313/ (bind address 0.0.0.0)
...
+```
+
+The tool will crawl your local website and will report if there's any broken link on it.
+
+If there's any, and you can't fix it right away, please [file an issue on GitHub](https://github.com/spiffe/spiffe.io/issues/new]).
```
Comment (sanderson042) [inline README.md:89]: Typo, extra character; slight clarification; combine paragraphs:

```suggestion
The tool will crawl your local website and report if there's any broken link on it. If there's any, and you can't create a PR to fix the link right away, please [file an issue on GitHub](https://github.com/spiffe/spiffe.io/issues/new).
```

File: README.md [hunk 5]
```diff
@@ -66,6 +66,28 @@ Web Server is available at //localhost:1313/ (bind address 0.0.0.0)
 
 The website is now available at [`http://localhost:1313`](http://localhost:1313). Changes in the Markdown files or the [external content descriptor file](./external.yaml) trigger a rebuild of the site. After the rebuild, the site is reloaded in your browser.
 
+### Avoiding broken links
```
Comment (sanderson042) [inline README.md:69]:
```suggestion
### Checking for broken links
```

**Review by sanderson042 — COMMENTED** [review]
Hi @mchurichi - there were a couple minor typos in the README. In the process of pointing those out in this review, I did a little editing, too. :-)

**Comment by mchurichi** [comment]
> Hi @mchurichi - there were a couple minor typos in the README. In the process of pointing those out in this review, I did a little editing, too. :-)

Thanks @sanderson042!, I've already pushed a commit with the changes you suggested, please take a look again.

**Review by sanderson042 — APPROVED** [review]
Looks good, thank you Max! Feel free to merge this PR when you get a chance.

**Pull request merged**